taskDev7/src/dateFormatChecker/dateChecker/HourChecker.java

```diff
@@ -0,0 +1,24 @@
+package dateFormatChecker.dateChecker;
+
+/**
+ * Checks that hour has right value.
+ */
+public class HourChecker extends DateChecker{
+
+  /**
+   * Checks that value of hour isn't out of borders (00-23).
+   *
+   * @param value value of hour which need to check.
+   * @return true if it satisfies the conditions.
+   */
+  @Override
+  public boolean check(String value) {
+    int hour = Integer.parseInt(value);
+
+    if (hour >= 0 && hour < 24 && value.length() == 2) {
```

Comment:
You can make some interval checker and drop all these classes with the same checks